File: software/examples/java/ExampleCallback.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletLinearPotiV2;

public class ExampleCallback {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Linear Poti Bricklet 2.0
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletLinearPotiV2 lp = new BrickletLinearPotiV2(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Add position listener
		lp.addPositionListener(new BrickletLinearPotiV2.PositionListener() {
			public void position(int position) {
				System.out.println("Position: " + position + " °");
			}
		});

		// Set period for position callback to 0.25s (250ms) without a threshold
		lp.setPositionCallbackConfiguration(250, false, 'x', 0, 0);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletLinearPotiV2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletLinearPotiV2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Linear Poti Bricklet 2.0

// Callback function for position callback
function cb_position($position)
{
    echo "Position: $position %\n";
}

$ipcon = new IPConnection(); // Create IP connection
$lp = new BrickletLinearPotiV2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register position callback to function cb_position
$lp->registerCallback(BrickletLinearPotiV2::CALLBACK_POSITION, 'cb_position');

// Set period for position callback to 0.25s (250ms) without a threshold
$lp->setPositionCallbackConfiguration(250, FALSE, 'x', 0, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
